lib/src/execute.rs

```diff
@@ -354,6 +357,57 @@ impl ExecuteCtx {
 
         outcome
     }
+
+    pub async fn run_main(self, args: &[String]) -> Result<(), anyhow::Error> {
+        // placeholders for request, result sender channel, and remote IP
+        let req = Request::get("http://example.com/").body(Body::empty())?;
+        let req_id = 0;
+        let (sender, _) = oneshot::channel();
+        let remote = Ipv4Addr::LOCALHOST.into();
+
+        let session = Session::new(
+            req_id,
+            req,
+            sender,
+            remote,
+            self.backends.clone(),
+            self.geolocation.clone(),
+            self.tls_config.clone(),
+            self.dictionaries.clone(),
+            self.config_path.clone(),
+            self.object_store.clone(),
+            self.secret_stores.clone(),
+        );
+
+        let mut store = create_store(&self, session).map_err(ExecutionError::Context)?;
+        store.data_mut().wasi().push_arg("wasm_program")?;
```

Comment:
Instead of `wasm_program`, we should use the filename of the module minus its extension. So `viceroy --run my_cool_app` should see `argv[0] == "my_cool_app"`.

Reply:
Since this is close enough to the core functionality of this PR, I'd prefer to address this one here before merging